### PR TITLE
Area effects

### PR DESCRIPTION
**Pull request by kulakowskin**
Implemented TakeDamage, HealDamage, InstantDeath, and LevelUp. Tested in GameState.

### CODE REVIEW

File: src/com/TigersIter2/areaEffects/AreaEffect.java [hunk 1]
```diff
@@ -10,8 +14,19 @@
  */
 public abstract class AreaEffect {
 
-    private Location location;
-    public StatsModifier statsMod;
+    private Location location;  //This is the location used by MODELS to determine where the areaEffect is
+    private Location pixelLocation; //This is the location used by VIEWS to determine where the areaEffect is
```
Comment (ahruss) [inline src/com/TigersIter2/areaEffects/AreaEffect.java:18]: Shouldn't this be a concern for the views to figure out? Why does the model need to know about its visual representation, or even that pixels exist?

File: src/com/TigersIter2/areaEffects/LevelUp.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import com.TigersIter2.stats.StatsModifier;
+import com.sun.xml.internal.bind.v2.TODO;
+
+/**
+ * Created by MyMac on 3/7/16.
+ */
+public class LevelUp extends AreaEffect{
+
+
+    public void affectEntity(Entity entity){
```
Comment (ahruss) [inline src/com/TigersIter2/areaEffects/LevelUp.java:15]: what if this is called with a non-avatar entity? For example, what if an NPC walks into the tile? This should check if the entity is an avatar before trying to do this

File: src/com/TigersIter2/managers/AreaEffectManager.java [hunk 3]
```diff
@@ -11,19 +14,38 @@
...
+            case "instantDeath": instantDeath.affectEntity(entityOnTile);
+                break;
+            case "takeDamage": takeDamage.affectEntity(entityOnTile);
+                break;
+            case "healDamage": healDamage.affectEntity(entityOnTile);
+                break;
+            case "levelUp": levelUp.affectEntity(entityOnTile);
+                break;
                 // add other cases for other area effects
+                // is teleport an area affect???
```
Comment (ahruss) [inline src/com/TigersIter2/managers/AreaEffectManager.java:48]: Why is this not done is such a way that it's possible to `do getAreaEffect().affectEntity(entity)`? In fact, if I understand correctly, it _is_  possible to just do `areaEffect.affectEntity(entityOnTile)`

File: src/com/TigersIter2/areaEffects/AreaEffect.java (hunk 1 above)
Comment (kulakowskin) [inline src/com/TigersIter2/areaEffects/AreaEffect.java:18]: you're right. pixelLocation isn't needed so I removed it.

File: src/com/TigersIter2/areaEffects/LevelUp.java (hunk 2 above)
Comment (kulakowskin) [inline src/com/TigersIter2/areaEffects/LevelUp.java:15]: Valid - fixed to first check that object type is Avatar

File: src/com/TigersIter2/managers/AreaEffectManager.java (hunk 3 above)
Comment (kulakowskin) [inline src/com/TigersIter2/managers/AreaEffectManager.java:48]: You're totally right. Not sure why I over complicated that. Thanks!

**Pull request merged**